src/tasks/solvers.ts

```diff
@@ -7,33 +7,47 @@ import { HardhatRuntimeEnvironment } from "hardhat/types";
 import { getDeployedContract } from "./ts/deployment";
 import { getNamedSigner } from "./ts/signers";
 import { getSolvers } from "./ts/solver";
+import { transactionUrl } from "./ts/tui";
 
 const solversTaskList = ["add", "check", "remove", "list"] as const;
 type SolversTasks = typeof solversTaskList[number];
 
-async function addSolver(solver: string, hre: HardhatRuntimeEnvironment) {
-  const owner = await getNamedSigner(hre, "manager");
-  const authenticator = await getDeployedContract(
-    "GPv2AllowListAuthentication",
-    hre,
-  );
+interface Args {
+  solver?: string;
+  printTransaction: boolean;
+}
 
-  const tx = await authenticator.connect(owner).addSolver(solver);
-  await tx.wait();
-  console.log("Solver added.");
+async function addSolver(args: Args, hre: HardhatRuntimeEnvironment) {
+  await performSolverManagement("addSolver", args, hre);
 }
 
-const removeSolver = async (solver: string, hre: HardhatRuntimeEnvironment) => {
-  const owner = await getNamedSigner(hre, "manager");
+const removeSolver = async (args: Args, hre: HardhatRuntimeEnvironment) => {
+  await performSolverManagement("removeSolver", args, hre);
+};
+
+async function performSolverManagement(
+  method: string,
```

Comment:
There is an assumption that the chosen method is callable with exactly one solver as an argument. I'd like to restrict `method` using the typing system if possible:
```suggestion
  method: "addSolver" | "removeSolver",
```